service/create/secrets.go

```diff
@@ -0,0 +1,72 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"k8s.io/client-go/pkg/api"
+	apierrs "k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/watch"
+)
+
+const (
+	componentLabel string        = "clusterComponent"
+	clusterIDLabel string        = "clusterID"
+	secretTimeOut  time.Duration = 10
+)
+
+func (s *Service) getCertsFromSecrets(clusterID string) (map[string]map[string][]byte, error) {
+	components := []string{
+		"apiserver",
+		"worker",
+		"etcd",
+		"calico",
+	}
+	assetTypes := []string{"crt", "key", "ca"}
+	// e.g. certs["worker"]["crt"
+	certs := make(map[string]map[string][]byte)
+
+	watcher, err := s.k8sClient.Core().Secrets(api.NamespaceDefault).Watch(v1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s in (%s), %s=%s", componentLabel, strings.Join(components, ","), clusterIDLabel, clusterID),
+	})
+	if err != nil {
+		return nil, microerror.MaskAny(err)
+	}
+
+	for i := 0; i < len(components); i++ {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return nil, microerror.MaskAny(fmt.Errorf("secrets channel was already closed"))
```

Comment:
I would like to use specific errors. Then we can do this. 

```
return nil, microerror.MaskAnyf(executionFailedError, "secrets channel was already closed")
```

That way we do not need `fmt`. We have typed errors we can check against. The formatting can also be done by `MaskAnyf`.